pic-sure-resources/pic-sure-aggregate-data-sharing-resource/src/main/java/edu/harvard/hms/dbmi/avillach/AggregateDataSharingResourceRS.java

```diff
@@ -234,35 +216,34 @@ public Response querySync(QueryRequest queryRequest) {
 			}
 			String expectedResultType = jsonNode.get("expectedResultType").asText();
 
-			if (! ALLOWED_RESULT_TYPES.contains(expectedResultType)) {
+			Set<String> allowedResultTypes =
+				Set.of("COUNT", "CROSS_COUNT", "INFO_COLUMN_LISTING", "OBSERVATION_COUNT", "OBSERVATION_CROSS_COUNT");
+			if (!allowedResultTypes.contains(expectedResultType)) {
 				logger.warn("Incorrect Result Type: " + expectedResultType);
 				return Response.status(Response.Status.BAD_REQUEST).build();
 			}
 
-			String targetPicsureUrl = properties.getTargetPicsureUrl();
-			String queryString = json.writeValueAsString(queryRequest);
+			String payload = json.writeValueAsString(queryRequest);
 			String pathName = "/query/sync";
-			String composedURL = composeURL(targetPicsureUrl, pathName);
-			logger.debug("Aggregate Data Sharing Resource, sending query: " + queryString + ", to: " + composedURL);
-			HttpResponse response = retrievePostResponse(composedURL, headers, queryString);
-			if (response.getStatusLine().getStatusCode() != 200) {
-				logger.error("Not 200 status!");
+			String composedURL = HttpClientUtil.composeURL(properties.getTargetPicsureUrl(), pathName);
+			logger.debug("Aggregate Data Sharing Resource, sending query: " + payload + ", to: " + composedURL);
+			HttpResponse response = HttpClientUtil.retrievePostResponse(composedURL, headers, payload);
+			if (HttpClientUtil.is2xx(response)) {
 				logger.error(
-						composedURL + " calling resource with id " + resourceUUID + " did not return a 200: {} {} ",
+						"{} calling resource with id {} did not return a 200: {} {} ", composedURL, resourceUUID,
 						response.getStatusLine().getStatusCode(), response.getStatusLine().getReasonPhrase());
-				throwResponseError(response, targetPicsureUrl);
+				HttpClientUtil.throwResponseError(response, properties.getTargetPicsureUrl());
```

Comment:
I prefer to qualify these static methods. One step closer to actual DI, and it makes it a bit easier to grok.